starknet_py/tests/e2e/account/account_test.py

```diff
@@ -381,20 +389,36 @@ async def test_deploy_account_raises_on_no_enough_funds(deploy_account_details_f
                 class_hash=class_hash,
                 salt=salt,
                 key_pair=key_pair,
-                client=GatewayClient(net="testnet"),
+                client=client,
                 chain=StarknetChainId.TESTNET,
                 max_fee=MAX_FEE,
             )
 
 
 @pytest.mark.asyncio
-async def test_deploy_account_passes_on_enough_funds(deploy_account_details_factory):
+@pytest.mark.parametrize(
+    "call_contract, deploy_account, client",
+    [
+        (
+            "starknet_py.net.gateway_client.GatewayClient.call_contract",
+            "starknet_py.net.gateway_client.GatewayClient.deploy_account",
+            "gateway_client",
+        ),
+        (
+            "starknet_py.net.full_node_client.FullNodeClient.call_contract",
+            "starknet_py.net.full_node_client.FullNodeClient.deploy_account",
+            "full_node_client",
+        ),
+    ],
+)
+async def test_deploy_account_passes_on_enough_funds(
+    deploy_account_details_factory, call_contract, deploy_account, client, request
+):
     address, key_pair, salt, class_hash = await deploy_account_details_factory.get()
+    client = request.getfixturevalue(client)
```

Comment:
Why do we need this here? Wouldn't it be enough to pass client fixture?

Reply:
Without it `client` fixture would be used as:
- FullNodeClient - full_node method
- FullNodeClient - gateway method
- GatewayClient - full_node method
- GatewayClient - gateway method.